plugins/device/rest.go

```diff
@@ -55,19 +54,19 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if _, exists := peerInfo.Metadata["_devices"]; exists {
-		var devices []deviceapi.Info
-		err = json.Unmarshal([]byte(peerInfo.Metadata["_devices"]), &devices)
-		if err != nil {
-			logger.WithError(err).WithField("peerid", peerID).Error(err)
-			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
-			return
-		}
-		if deviceutils.DeviceExist(req.Device, devices) {
-			logger.WithError(err).WithField("device", req.Device).Error("Device already exists")
-			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Device already exists")
-			return
-		}
+	_, exists := peerInfo.Metadata["_devices"]
+	devices, err := deviceutils.FetchDevices(peerInfo)
+
+	if devices == nil && exists && err != nil {
+		logger.WithError(err).WithField("peerid", peerID).Error(err)
```

Comment:
This log entry doesn't have a message. The `err` should be passed to `WithError` and is not a substitute for message.